backend/kernelCI_app/views/hardwareDetailsBootsView.py

```diff
@@ -130,6 +131,14 @@ def post(self, request, hardware_id) -> Response:
             end_datetime=self.end_datetime,
         )
 
+        if len(trees) == 0:
+            return create_api_error_response(
+                error_message={
+                    "error": "This hardware isn't associated with any commit"
+                },
```

Comment:
`create_api_error_response` already returns `{"error": error_message}`. You should pass only the message, not the whole object here
```suggestion
            return create_api_error_response(
                error_message="This hardware isn't associated with any commit",
```